src/test/java/IntegrationTests/WAStorageClientDownloadIT.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package IntegrationTests;
+
+import com.microsoft.azure.storage.AccessCondition;
+import com.microsoft.azure.storage.CloudStorageAccount;
+import com.microsoft.azure.storage.StorageCredentialsAccountAndKey;
+import com.microsoft.azure.storage.blob.CloudBlobClient;
+import com.microsoft.azure.storage.blob.CloudBlobContainer;
+import com.microsoft.azure.storage.blob.CloudBlockBlob;
+import com.microsoftopentechnologies.windowsazurestorage.WAStorageClient;
+import com.microsoftopentechnologies.windowsazurestorage.beans.StorageAccountInfo;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import java.io.File;
+import java.io.IOException;
+import java.util.UUID;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Test;
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+
+/**
+ *
+ * @author arroyc
+ */
+public class WAStorageClientDownloadIT extends IntegrationTest{
+    
+    private static final Logger LOGGER = Logger.getLogger(WAStorageClientDownloadIT.class.getName());
+    
+    /**
+     * Test of validateStorageAccount method, of class WAStorageClient.
+     */
+    @Test
+    public void testValidateStorageAccount() throws Exception {
+        System.out.println("validateStorageAccount");
+        StorageAccountInfo storageAccount = testEnv.sampleStorageAccount;
+        boolean expResult = true;
+        boolean result = WAStorageClient.validateStorageAccount(storageAccount);
+        assertEquals(expResult, result);
+    }
+
+    @Test(timeout=1200000)
+    public void testDownloadfromContainer() {
+        System.out.print("download");
+        CloudBlobContainer container;
+        try{
+            CloudStorageAccount account = new CloudStorageAccount(new StorageCredentialsAccountAndKey(testEnv.azureStorageAccountName, testEnv.azureStorageAccountKey1));
+            CloudBlobClient blobClient = account.createCloudBlobClient();
+            String containerName = "testdownload"+TestEnvironment.GenerateRandomString(15);
+            File directory = new File(containerName);
+            directory.mkdir();
+            container = blobClient.getContainerReference(containerName);
+            container.deleteIfExists();
+            container.createIfNotExists();
+            for(int i=0; i<testEnv.TOTAL_FILES;i++){
+                testEnv.downloadFileList.add(new File(directory, "download"+UUID.randomUUID().toString()+".txt"));
```

Comment:
this part seems very similar with the upload. I would move it in a method inside IntegrationTest class